### PR TITLE
Rename `burned_mana` to `max_burned_mana`

### LINKED ISSUE #1351: Rename BurnedMana to MaxBurnedMana
https://github.com/iotaledger/iota.go/pull/538

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

Rename this field to fit incoming TIP.

## Links to any relevant issues

Closes #1351

### CODE REVIEW

File: sdk/src/types/block/core/basic.rs [hunk 1]
```diff
@@ -140,7 +140,7 @@ impl BasicBlock {
     /// Returns the burned mana of a [`BasicBlock`].
     #[inline(always)]
     pub fn burned_mana(&self) -> u64 {
```
Comment (thibault-martinez) [inline sdk/src/types/block/core/basic.rs]: Rename + adapt comment

Comment (DaughterOfMars) [reply sdk/src/types/block/core/basic.rs]: done

Comment (thibault-martinez) [reply sdk/src/types/block/core/basic.rs]: Function hasn't been renamed ?

Comment (DaughterOfMars) [reply sdk/src/types/block/core/basic.rs]: I don't understand how this happened

**Review by thibault-martinez — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**